x-pack/plugins/lens/public/indexpattern_datasource/operations/layer_helpers.ts

```diff
@@ -864,18 +864,24 @@ export function updateLayerIndexPattern(
  * - All column references are valid
  * - All prerequisites are met
  */
-export function getErrorMessages(layer: IndexPatternLayer): string[] | undefined {
-  const errors: string[] = [];
-  Object.entries(layer.columns).forEach(([columnId, column]) => {
-    // If we're transitioning to another operation, check for "new" incompleteColumns rather
-    // than "old" saved operation on the layer
-    const columnFinalRef =
-      layer.incompleteColumns?.[columnId]?.operationType || column.operationType;
-    const def = operationDefinitionMap[columnFinalRef];
-    if (def.getErrorMessage) {
-      errors.push(...(def.getErrorMessage(layer, columnId) ?? []));
-    }
-  });
+export function getErrorMessages(
+  layer: IndexPatternLayer,
+  indexPattern?: IndexPattern
+): string[] | undefined {
+  const errors: string[] = Object.entries(layer.columns)
+    .flatMap(([columnId, column]) => {
```

Comment:
So in terms of style, you're using `flatMap` and then `filter` as opposed to `forEach`- seems fine.

Reply:
I was just hoping to use a simple `.filter(Boolean)` at the end, but Typescript wasn't happy about that., therefore I had to god for an explicit typeguard.